src/classes/modules/java.base/java/lang/String.java

```diff
@@ -241,6 +241,14 @@ public char charAt(int index) {
 		return StringCoding.encode(x, coder(), value);
 	}
 
+	void getBytes(byte dst[], int dstBegin, byte coder) {
+		if (coder() == coder) {
+			System.arraycopy(value, 0, dst, dstBegin << coder, value.length);
+		} else {
+			StringLatin1.inflate(value, 0, dst, dstBegin, value.length);
```

Comment:
This seems to cover two cases:

1. Both encodings (source, target) are the same.
2. The target encoding is UTF-8 but the source is Latin-1.

However, what about the case that the source encoding is UTF-8 and the target is Latin-1?

Reply:
The official doc comment include this:

> Invoker guarantees: dst is in UTF16 (inflate itself for asb), if two coders are different, and dst is big enough (range check)

The method is also package-private. Anyway I could add an assertion that checks whether the target is in UTF16.